pkg/controllers/raycluster_webhook_test.go

```diff
@@ -0,0 +1,668 @@
+							Containers: []corev1.Container{
+								{
+									Name: "worker-container-1",
+									Env: []corev1.EnvVar{
+										{
+											Name: "MY_POD_IP",
+											ValueFrom: &corev1.EnvVarSource{
+												FieldRef: &corev1.ObjectFieldSelector{
+													FieldPath: "status.podIP",
+												},
+											},
+										},
+										{Name: "RAY_USE_TLS", Value: "1"},
+										{Name: "RAY_TLS_SERVER_CERT", Value: "/home/ray/workspace/tls/server.crt"},
+										{Name: "RAY_TLS_SERVER_KEY", Value: "/home/ray/workspace/tls/server.key"},
+										{Name: "RAY_TLS_CA_CERT", Value: "/home/ray/workspace/tls/ca.crt"},
+									},
+								},
+							},
+							InitContainers: []corev1.Container{
+								{
+									Name:  "create-cert",
+									Image: "",
+									Command: []string{
+										"sh",
+										"-c",
+										`cd /home/ray/workspace/tls && openssl req -nodes -newkey rsa:2048 -keyout server.key -out server.csr -subj '/CN=ray-head' && printf "authorityKeyIdentifier=keyid,issuer\nbasicConstraints=CA:FALSE\nsubjectAltName = @alt_names\n[alt_names]\nDNS.1 = 127.0.0.1\nDNS.2 = localhost\nDNS.3 = ${FQ_RAY_IP}\nDNS.4 = $(awk 'END{print $1}' /etc/hosts)">./domain.ext && cp /home/ray/workspace/ca/* . && openssl x509 -req -CA ca.crt -CAkey ca.key -in server.csr -out server.crt -days 365 -CAcreateserial -extfile domain.ext`,
+									},
+									VolumeMounts: certVolumeMounts(),
+								},
+							},
+							Volumes: []corev1.Volume{
+								{
+									Name: "ca-vol",
+									VolumeSource: corev1.VolumeSource{
+										Secret: &corev1.SecretVolumeSource{
+											SecretName: `ca-secret-` + rayClusterName,
+										},
+									},
+								},
+								{
+									Name: "server-cert",
+									VolumeSource: corev1.VolumeSource{
+										EmptyDir: &corev1.EmptyDirVolumeSource{},
+									},
+								},
+							},
+						},
+					},
+					RayStartParams: map[string]string{},
+				},
+			},
+		},
+	}
+
+	// Create the RayClusters
+	if _, err := test.Client().Ray().RayV1().RayClusters(namespace).Create(test.Ctx(), validRayCluster, metav1.CreateOptions{}); err != nil {
+		test.T().Fatalf("Failed to create RayCluster: %v", err)
+	}
+
+	// Positive Test Case: Valid RayCluster with immutable fields
+	t.Run("Expected no warnings or errors on call to ValidateUpdate function", func(t *testing.T) {
+		warnings, err := rcWebhook.ValidateUpdate(test.Ctx(), runtime.Object(validRayCluster), runtime.Object(validRayCluster))
+		test.Expect(warnings).Should(BeNil(), "Expected no warnings on call to ValidateUpdate function")
+		test.Expect(err).ShouldNot(HaveOccurred(), "Expected no errors on call to ValidateUpdate function")
+	})
+
+	// Negative Test Cases
+	trueBool := true
+	invalidRayCluster := validRayCluster.DeepCopy()
+
+	t.Run("Negative: Expected errors on call to ValidateUpdate function due to EnableIngress set to True", func(t *testing.T) {
+		invalidRayCluster.Spec.HeadGroupSpec.EnableIngress = &trueBool
```

Comment:
Nitpick, there is a dedicate funtion to create pointer for values:
```suggestion
		invalidRayCluster.Spec.HeadGroupSpec.EnableIngress = support.Ptr(true)
```